lib/dialects/mssql/query/compiler.js

```diff
@@ -186,17 +186,31 @@ class QueryCompiler_MSSQL extends QueryCompiler {
 
   _buildReturningDeclare(values) {
     if (values && values.length > 0) {
-      const columns = values.map(value => `[${value}] nvarchar(max) null`);
-      return `declare @out table (${columns.join(",")}); `;
+      let columns = "";
+
+      if (Array.isArray(values)) {
+        columns = values.map(value => `[${value}] nvarchar(max) null`).join(",");
+      } else {
+        columns = `[${values}] nvarchar(max) null`
+      }
+
+      return `declare @out table (${columns}); `;
     }
 
     return '';
   }
 
   _buildReturningSelect(values) {
     if (values && values.length > 0) {
-      const selections = values.map(value => `[${value}]`);
-      return `; select ${selections.join(",")} from @out`
+      let selections = "";
+
+      if (Array.isArray(values)) {
+        selections = values.map(value => `[${value}]`).join(",");
```

Comment:
will this work with multiple values? it is supposed to be `[xyz], [abc]`?

Reply:
From what I saw when I debugged this 2 types of values came in ("<data>" and ["data1","data2",...]). This will work with both of these use cases and is tested in multiple pre-existing test cases.

Reply:
looooots of tests are failing now, though.

Reply:
It is failing because the test is using a string comparison of the generated SQL to validate its responses. This no long equals the generated sql because of the added functionality.  Would you like me to go into each test that is failing using string comparisons and update them accordingly?

Note there are a large number of tests that would need to be changed and might take some time to finalize. Wanted to get you're okay before doing this since this is touching tests that are not a part of my change directly.

Reply:
So end result is still the same, just SQL is slightly different?
Yeah, unfortunately I don't think it is possible to get CI to green without changing the tests :(

Reply:
Okay. It took awhile but I got the logic modified to handle returning "*" and modified all existing tests that only test the generated sql string to check for the new format.